tests/profiles_enabled/shipments/test_access_requests.py

```diff
@@ -0,0 +1,637 @@
+from django.urls import reverse
+from pytest import fixture
+from shipchain_common.test_utils import AssertionHelper
+
+from apps.shipments.models import AccessRequest, PermissionLevel, Endpoints, TrackingData
+
+
+@fixture
+def access_request_ro_attributes():
+    return {
+        'shipment_permission': PermissionLevel.READ_ONLY.name,
+        'tags_permission': PermissionLevel.READ_ONLY.name,
+        'documents_permission': PermissionLevel.READ_ONLY.name,
+        'notes_permission': PermissionLevel.READ_ONLY.name,
+        'tracking_permission': PermissionLevel.READ_ONLY.name,
+        'telemetry_permission': PermissionLevel.READ_ONLY.name,
+    }
+
+@fixture
+def access_request_rw_attributes():
+    return {
+        'shipment_permission': PermissionLevel.READ_WRITE.name,
+        'tags_permission': PermissionLevel.READ_WRITE.name,
+        'documents_permission': PermissionLevel.READ_WRITE.name,
+        'notes_permission': PermissionLevel.READ_WRITE.name,
+        'tracking_permission': PermissionLevel.READ_ONLY.name,
+        'telemetry_permission': PermissionLevel.READ_ONLY.name,
+    }
+
+
+@fixture
+def new_access_request_bob(shipment_alice, user_bob, access_request_ro_attributes):
+    return AccessRequest.objects.create(shipment=shipment_alice, requester_id=user_bob.id, **access_request_ro_attributes)
+
+@fixture
+def new_rw_access_request_bob(shipment_alice, user_bob, access_request_rw_attributes):
+    return AccessRequest.objects.create(shipment=shipment_alice, requester_id=user_bob.id, **access_request_rw_attributes)
+
+@fixture
+def approved_access_request_bob(new_access_request_bob):
+    new_access_request_bob.approved = True
+    new_access_request_bob.save()
+    return new_access_request_bob
+
+@fixture
+def new_access_request_lionel(shipment_alice, user_lionel, access_request_ro_attributes):
+    return AccessRequest.objects.create(shipment=shipment_alice, requester_id=user_lionel.id, **access_request_ro_attributes)
+
+@fixture
+def approved_access_request_lionel(new_access_request_lionel):
+    new_access_request_lionel.approved = True
+    new_access_request_lionel.save()
+    return new_access_request_lionel
+
+@fixture
+def entity_ref_access_request(new_access_request_bob, user_bob_id, entity_ref_shipment_alice):
```

Comment:
Good call. I think I wrote this fixture and then forgot about it. I'll either remove it or find other places to use it (if necessary).

Reply:
you test a list and ensure that `count=1` in it when testing access to the `access-request` list endpoints. You could use this fixture in those instances, but wouldn't be as necessary since you're not testing that a specific `access_request` object was returned.